Makefile

```diff
@@ -18,12 +18,12 @@ BACKTRACE=RUST_BACKTRACE=1
 
 .PHONY: clippy
 clippy: ## Runs Clippy with configs
-	cargo clippy --workspace --all-targets $(ALL_FEATURES_BUT_ASYNC) -- -D warnings
+	${BUILD_GENERATED_FILES_IN_SRC} cargo clippy --workspace --all-targets $(ALL_FEATURES_BUT_ASYNC) -- -D warnings
 
 
 .PHONY: clippy-no-std
 clippy-no-std: ## Runs Clippy with configs
-	cargo clippy --no-default-features --target wasm32-unknown-unknown --workspace --lib --features tx-prover --exclude miden-proving-service -- -D warnings
+	${BUILD_GENERATED_FILES_IN_SRC} cargo clippy --no-default-features --target wasm32-unknown-unknown --workspace --lib --features tx-prover --exclude miden-proving-service -- -D warnings
```

Comment:
Why do we run clippy with this flag on but don't do this for other commands (e.g., `fix` or `format`)?

If possible, I'd probably enable this flag just for the `lint` command (but not sure if it'll work since `lint` is a "composite" command).

Reply:
I ended up using this approach:
```
lint: ## Runs all linting tasks at once (Clippy, fixing, formatting)
	@$(BUILD_GENERATED_FILES_IN_SRC) $(MAKE) format
	@$(BUILD_GENERATED_FILES_IN_SRC) $(MAKE) fix
	@$(BUILD_GENERATED_FILES_IN_SRC) $(MAKE) clippy
	@$(BUILD_GENERATED_FILES_IN_SRC) $(MAKE) clippy-no-std
```

I first tried with a target to export the env var, but it was adding noise when running `make`.